mne/time_frequency/psd.py

```diff
@@ -145,6 +145,10 @@ def psd_array_welch(x, sfreq, fmin=0, fmax=np.inf, n_fft=256, n_overlap=0,
     .. footbibliography::
     """
     _check_option('average', average, (None, False, 'mean', 'median'))
+    _validate_type(n_fft, "int", "n_fft")
+    _validate_type(n_overlap, "int", "n_overlap")
+    if n_per_seg is not None:
+        _validate_type(n_per_seg, "int", "n_per_seg")
```

Comment:
```suggestion
    _validate_type(n_per_seg, ("int", None), "n_per_seg")
```

Reply:
Actually, it does not support multiple types if one of them is `"int"`

Reply:
https://github.com/mne-tools/mne-python/blob/063a3e87c521fc0bff0307a456cb8d363ecffa9e/mne/utils/check.py#L517-L519

Reply:
Are you sure? To me it’s mostly to speed up the check but it should still work

Reply:
Sure, the function fails with a KeyError for "int"

Reply:
https://github.com/mne-tools/mne-python/actions/runs/3507673189/jobs/5875539030

But I agree it should have work.

Reply:
After the is-not-None check I would use _ensure_int, which casts for you

Reply:
@larsoner Does it change anything? `n_fft  = _ensute_int(n_fft, "n_fft")` or `_validate_type(n_fft, "int")` will both raise if the argument is not an integer, e.g. a float `n_fft = 256.0`. It's not more forgiving and does not convert to an integer if `x == int(x)`.

Reply:
It should always return an actual python int, so I'm surprised about that last statement. But maybe I'm misremembering

Reply:
It probably converts from `np.int` to regular integers, but beside that,

```
from mne.utils import _ensure_int
_ensure_int(256.)
```

raises `TypeError: unknown must be an int, got <class 'float'>`.

Reply:
That looks like correct/useful behavior

Reply:
... @mscheltienne I think we are digressing a bit here. In MNE-Python we try to use `_ensure_int` when we want to ensure a user is passing an `int`-ike, e.g.:

```
$ git grep " = _ensure_int"
mne/_ola.py:        n_pts = _ensure_int(n_pts, 'n_pts')
mne/_ola.py:        n_samples = _ensure_int(n_samples, 'n_samples')
mne/_ola.py:        n_overlap = _ensure_int(n_overlap, 'n_overlap')
mne/_ola.py:        n_total = _ensure_int(n_total, 'n_total')
mne/bem.py:    threshold = _ensure_int(threshold, 'threshold')
mne/filter.py:    filter_length = _ensure_int(filter_length, 'filter_length')
mne/io/pick.py:        info = _ensure_int(info, 'info', 'an int or Info')
...
```
It sounds like that is your goal here, yes? If so, I think we should continue using `_ensure_int` here, and @mscheltienne if you are still opposed to this function / don't see the point / think it's useless, we should debate its merits in a separate issue that addresses that.

Reply:
I am not opposed in the slightest, especially on a minor point like this one, since both functions fulfill the same role. I am however curious, am I missing something about `_ensure_int`? `_validate_type` calls it internally, and the casting seems to occur only from int to int (otherwise it raises).

Reply:
> the casting seems to occur only from int to int

Based on this phrasing perhaps what you miss is that more specifically "the casting occurs from *any int-like that we want to support*, to *a native Python int*".

Why is this useful? Well here is one problematic example case:
```
>>> _validate_type(np.uint32(1), 'int')
>>> np.uint32(1) - np.uint32(2)
<stdin>:1: RuntimeWarning: overflow encountered in uint_scalars
4294967295
>>> _ensure_int(np.uint32(1)) - np.uint32(2)
-1
```
This may seem contrived, but once users start working with / saving data to different formats (and we use I/O internally, too, internally), this sort of stuff can happen, and motivated implementing this function. So can this:
```
>>> type(np.array(1))
<class 'numpy.ndarray'>
>>> type(_ensure_int(np.array(1)))
<class 'int'>
```
I can't remember offhand why having this 0-d `ndarray` was problematic downstream, but I think it was at some point...

And even for less problematic cases (e.g., `np.int32` vs `np.int64` for small values), having different variants of int-like variables being passed can end up being inefficient. For example, if you write a numba function that consumes it, numba will now have to JIT-compile a new variant for every dtype (or raise if it doesn't know how to handle the inputs).

So anything that requires, is made safer, or is made more consistent by knowing that what you work with is a native Python `int` will benefit from this function. And given that it's hard to see ahead of time how potential failures will come up and account for them, it's useful to use it most times you need an int.

Reply:
Thank you for the details, I'll think about it and about how to include this in my future codes. I did not realize this difference between ints could cause issues.